src/Engine.h

```diff
@@ -5,6 +5,15 @@
 #include <vector>
 #include <map>
 
+#include <v8-isolate.h>

```

Comment:
@devm18426 do you think including the header files just if compiling on Windows could make sense?

Reply:
I don't mind adding some OS checks but I have to ask: How does this work on Linux?
Are these files being implicitly included somehow with gcc? They're definitely required for the code to build from what I can tell.
And if they are being implicitly included, what would be the reasoning behind hiding these explicit includes?
Like I said, I don't mind either way. Just curious.

Reply:
Just asking because I never needed those explicit includes on Linux and macOS. Taking a look at [1] it seems like they are already included by v8.h. At the same time I remember you mentioning they were required for successfully building on Window so I was just curious about that...

[1} https://github.com/v8/v8/blob/main/include/v8.h

Reply:
When investigating I noticed that as well however the build still wouldn't succeed. I did think it was weird at the time. It could be that there were other pending issues (maybe linker related) that hadn't been addressed yet. I'll double check to make sure.

Reply:
@buffer I was on the verge of giving up after going down a million rabbit holes. I couldn't for the life of me figure this out. [Here](https://github.com/devm18426/stpyv8/actions/runs/5936702344/job/16097991470) you may find the errors that arise when compiling without the explicit includes. They're bizarre, nonsensical and meaningless.

Googling "V8 C2143 C2059" lead to a few relevant SO questions but none held a direct golden solution. [This](https://stackoverflow.com/questions/9567868/compile-errors-after-adding-v8-to-my-project-c2143-c2059) post gave me a hunch there's some buggery going on during the preprocess stage (would explain the unhelpful error messages). But I couldn't figure it out. I tried everything. Played with the compiler flags, recompiled v8 using multiple configurations, nothing worked.

That is until I moved `#include <v8.h>` above any Boost includes.

I believe this is the key to everything. A [workflow](https://github.com/devm18426/stpyv8/actions/runs/5941319935) is currently checking whether this is a solution. We'll know for sure in about 15 minutes.

If I had to guess why this works, I'd say Boost (which has become my #1 least favorite piece of software over the past few days) sets a few `#define`s behind the scenes which completely screw up the preprocessor context. This is what dooms v8 when `#include`ing afterwards. I don't know what the specific "bad" operations are so I can't attempt to provide a more elegant fix. This problem only seems to arise when Boost, v8 and Windows are all mixed together...

Would you be ok maintaining your code from now on such that v8 is always `#include`d before Boost?

Reply:
Yep that did the trick... So we don't need explicit includes, just for v8 to come before Boost.

Please let me know what you think.

Reply:
I think moving the includes order should be not a big deal in the end. I agree with you that some defines are potentially redefined on Windows. I am going to release a new version of STPyV8 in the next hours. Soon after I will consider merging the PR so that we can start testing all the changes in the main branch